samcli/commands/delete/delete_context.py

```diff
@@ -0,0 +1,163 @@
+        self.s3_uploader = None
+        # self.uploaders = None
+        self.cf_template_file_name = None
+        self.delete_artifacts_folder = None
+        self.delete_cf_template_file = None
+
+    def __enter__(self):
+        self.parse_config_file()
+        if not self.stack_name:
+            self.stack_name = prompt(
+                f"\t{self.start_bold}Enter stack name you want to delete{self.end_bold}", type=click.STRING
+            )
+
+        if not self.region:
+            self.region = prompt(
+                f"\t{self.start_bold}Enter region you want to delete from{self.end_bold}", type=click.STRING
+            )
+        return self
+
+    def __exit__(self, *args):
+        pass
+
+    def parse_config_file(self):
+        """
+        Read the provided config file if it exists and assign the options values.
+        """
+        toml_provider = TomlProvider(CONFIG_SECTION, [CONFIG_COMMAND])
+        config_options = toml_provider(
+            config_path=self.config_file, config_env=self.config_env, cmd_names=[CONFIG_COMMAND]
+        )
+        if config_options:
+            if not self.stack_name:
+                self.stack_name = config_options.get("stack_name", None)
+            if self.stack_name == config_options["stack_name"]:
+                if not self.region:
+                    self.region = config_options.get("region", None)
+                if not self.profile:
+                    self.profile = config_options.get("profile", None)
+                self.s3_bucket = config_options.get("s3_bucket", None)
+                self.s3_prefix = config_options.get("s3_prefix", None)
+
+    def delete(self):
+        """
+        Delete method calls for Cloudformation stacks and S3 and ECR artifacts
+        """
+        template_str = self.cf_utils.get_stack_template(self.stack_name, TEMPLATE_STAGE)
+
+        # template_dict = yaml_parse(template_str)
+
+        if self.s3_bucket and self.s3_prefix:
+            self.delete_artifacts_folder = confirm(
+                f"\t{self.start_bold}Are you sure you want to delete the folder"
+                + f" {self.s3_prefix} in S3 which contains the artifacts?{self.end_bold}",
+                default=False,
+            )
+            if not self.delete_artifacts_folder:
+                with mktempfile() as temp_file:
+                    self.cf_template_file_name = get_cf_template_name(temp_file, template_str, "template")
+                self.delete_cf_template_file = confirm(
+                    f"\t{self.start_bold}Do you want to delete the template file"
+                    + f" {self.cf_template_file_name} in S3?{self.end_bold}",
+                    default=False,
+                )
+
+        click.echo("\n")
+        # Delete the primary stack
+        self.cf_utils.delete_stack(self.stack_name)
+
+        click.echo("- deleting Cloudformation stack {0}".format(self.stack_name))
+
+        # Delete the artifacts
+        # Intentionally commented
```

Comment:
Why?

Reply:
I had kept it as the starting code for the next task which is now on a different branch, but I can remove the commented code in this branch to keep it clean.